ghcjs-src/Miso/FFI.hs

```diff
@@ -101,23 +104,41 @@ foreign import javascript unsafe "console.log($1);"
 foreign import javascript unsafe "$r = JSON.stringify($1);"
   stringify' :: JSVal -> IO JSString
 
-foreign import javascript unsafe "$r = JSON.parse($1);"
+-- | Parse a raw string into JSON. Accepts any JS value.
+foreign import javascript unsafe "$r = JSON.parse('' + $1);"
```

Comment:
This seems fishy, when is this needed?

Reply:
So the problem there is that `JSON.parse()` expects a string, but the type signature for `parse'` accepts any `JSVal`. So you can get obscure exceptions thrown.

Admittedly the better solution would be to refactor the code so that `parse'` takes ` JSString` instead, and/or to wrap the `JSON.parse()` in a `try`/`catch` so that we can handle a JSON parse exception. I was actually planning on doing that but didn't get there because I didn't want this PR to get too large. I could certainly revert this change, but on the other hand, it doesn't make the code any *less* safe than it already is.